diff --git a/src/ukws_helper.h b/src/ukws_helper.h
index 610e91c..59806de 100644
--- a/src/ukws_helper.h
+++ b/src/ukws_helper.h
@@ -20,10 +20,8 @@
 #ifndef UKWS_HELPER_H
 #define UKWS_HELPER_H
 
-extern "C" {
 #include <glib.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
-}
 
 #include <QImage>
 #include <QPixmap>
diff --git a/src/ukws_indicator.cpp b/src/ukws_indicator.cpp
index 65805df..f10f857 100644
--- a/src/ukws_indicator.cpp
+++ b/src/ukws_indicator.cpp
@@ -34,11 +34,9 @@
 #include <QTime>
 #include <QCursor>
 
-extern "C" {
 #include <X11/X.h>
 #include <X11/Xlib.h>
 #include <unistd.h>
-}
 
 UkwsIndicator::UkwsIndicator(QWidget *parent) : QWidget(parent)
 {
diff --git a/src/ukws_manager.cpp b/src/ukws_manager.cpp
index a16786d..371b651 100644
--- a/src/ukws_manager.cpp
+++ b/src/ukws_manager.cpp
@@ -26,9 +26,7 @@
 #include <QX11Info>
 #include <QTime>
 
-extern "C" {
 #include <X11/Xlib.h>
-}
 
 //#define NOT_REG_WINDOW_SWITCH_HOTKEY
 //#define NOT_REG_WORKSPACE_VIEW_HOTKEY
diff --git a/src/ukws_window_box.h b/src/ukws_window_box.h
index feb6a4b..d16da93 100644
--- a/src/ukws_window_box.h
+++ b/src/ukws_window_box.h
@@ -20,10 +20,8 @@
 #ifndef UKWS_WINDOWBOX_H
 #define UKWS_WINDOWBOX_H
 
-extern "C" {
 #define WNCK_I_KNOW_THIS_IS_UNSTABLE
 #include <libwnck/libwnck.h>
-}
 
 #include "ukws_window_extra_label.h"
 
diff --git a/src/ukws_window_info.h b/src/ukws_window_info.h
index 2c6d4e6..ef590e8 100644
--- a/src/ukws_window_info.h
+++ b/src/ukws_window_info.h
@@ -20,10 +20,8 @@
 #ifndef UKWS_WINDOW_INFO_H
 #define UKWS_WINDOW_INFO_H
 
-extern "C" {
 #define WNCK_I_KNOW_THIS_IS_UNSTABLE
 #include <libwnck/libwnck.h>
-}
 
 #include <QObject>
 #include <QLabel>
diff --git a/src/ukws_wnck_operator.h b/src/ukws_wnck_operator.h
index 6391ad1..d75409e 100644
--- a/src/ukws_wnck_operator.h
+++ b/src/ukws_wnck_operator.h
@@ -20,10 +20,8 @@
 #ifndef UKWS_WNCK_OPERATOR_H
 #define UKWS_WNCK_OPERATOR_H
 
-extern "C" {
 #define WNCK_I_KNOW_THIS_IS_UNSTABLE
 #include <libwnck/libwnck.h>
-}
 
 #include <QList>
 
diff --git a/src/ukws_workspace_box.h b/src/ukws_workspace_box.h
index 1023ca7..a039e06 100644
--- a/src/ukws_workspace_box.h
+++ b/src/ukws_workspace_box.h
@@ -20,11 +20,9 @@
 #ifndef UKWSWORKSPACEBOX_H
 #define UKWSWORKSPACEBOX_H
 
-extern "C" {
 #define WNCK_I_KNOW_THIS_IS_UNSTABLE
 #include <libwnck/libwnck.h>
 #include <xcb/xcb.h>
-}
 
 #include "ukws_window_extra_label.h"
 #include "ukws_wnck_operator.h"
